kernel/src/fs/ramfs/fs.rs

```diff
@@ -519,7 +541,11 @@ impl RamInode {
 impl PageCacheBackend for RamInode {
     fn read_page_async(&self, _idx: usize, frame: &Frame) -> Result<BioWaiter> {
         // Initially, any block/page in a RamFs inode contains all zeros
-        frame.writer().fill(0);
+        frame
+            .writer()
+            .to_fallible()
+            .fill_zeros(frame.size())
```

Comment:
This change has nothing to do with optimizing dir entries, so it should be done in a separate commit.